Test for changing everything needed

Need a test to check that all desired fields could be changed via admin.

It may look like this:
```
def test_change_fields(self):
     r = self.client.get(self.change_url)
     data = self.get_form_data_from_response(r)
     changes = self.change_fields(data)
     data.update(changes)
     r = self.client.post(url, data=data)
     obj = self.get_object()
     for k, v in changes.items():
        self.assertEqual(getattr(obj, k), v)
def change_fields(data: dict) -> dict:
    raise NotImplementedError()
````

https://travis-ci.org/github/just-work/django-admin-blame/jobs/674282585

```
======================================================================
ERROR: setUpClass (admin_smoke.tests.AdminTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/django/test/testcases.py", line 1137, in setUpClass
    cls.setUpTestData()
  File "/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/admin_smoke/tests.py", line 44, in setUpTestData
    cls.admin = cls.model_admin(cls.model, site)
AttributeError: type object 'AdminTests' has no attribute 'model_admin'
```

AdminTests mixin is now discovered as a test case, it's a bug.

```
Traceback (most recent call last):
  File "/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/admin_smoke/tests.py", line 286, in test_all_fields_present
    f'fields {list(absent_fields)} are absent on form.')
AssertionError: {'modified', 'modified_by', 'created', 'created_by'} is not false : fields ['modified', 'modified_by', 'created', 'created_by'] are absent on form.
```

These fields are marked in model as non-editable, this should be considered as valid behavior.

--- a/admin_smoke/tests.py
+++ b/admin_smoke/tests.py
@@ -1,6 +1,6 @@
 from datetime import timedelta
 from typing import (Any, TypeVar, Type, Union, Optional, Iterable,
-                    cast, Dict, List)
+                    cast, Dict, List, TYPE_CHECKING, ClassVar)
 
 from django.contrib.admin import ModelAdmin, site
 from django.contrib.auth import get_user_model
@@ -24,13 +24,15 @@ M = TypeVar('M', bound=models.Model)
 
 class AdminBaseTestCase(BaseTestCase):
     """ Base class for django admin smoke tests."""
-    model_admin: Type[ModelAdmin]
-    model: Type[models.Model]
-    object_name: str
+    model_admin: ClassVar[Type[ModelAdmin]]
+    model: ClassVar[Type[models.Model]]
+    object_name: ClassVar[str]
+    excluded_fields: ClassVar[List[str]] = []
+    # model fields omitted in model admin
+
     changelist_url: str
     add_url: str
     admin: ModelAdmin
-    excluded_fields: List[str] = [] #  fields excluded from presence check
 
     @classproperty
     def opts(self) -> Options:
@@ -178,8 +180,14 @@ class AdminBaseTestCase(BaseTestCase):
         return data
 
 
+if TYPE_CHECKING:  # pragma: no cover
+    CommonAdminTestsTarget = AdminBaseTestCase
+else:
+    CommonAdminTestsTarget = object
+
+
 # noinspection PyAbstractClass
-class CommonAdminTests(AdminBaseTestCase):
+class CommonAdminTests(CommonAdminTestsTarget):
     """ Common smoke tests for django admin."""
 
     def test_changelist(self) -> None:
@@ -279,7 +287,7 @@ class CommonAdminTests(AdminBaseTestCase):
         form: ModelForm = cd['adminform'].form
         model_fields = {f.name for f in self.opts.fields
                         if f.name not in self.excluded_fields
-                        and not f.primary_key}
+                        and not f.primary_key or not f.editable}
         form_fields = set(form.Meta.fields)
         absent_fields = model_fields - form_fields
         self.assertFalse(absent_fields,

--- a/testproject/testapp/tests.py
+++ b/testproject/testapp/tests.py
@@ -4,7 +4,7 @@ from admin_smoke import tests
 from testproject.testapp import admin, models
 
 
-class ProjectAdminTestCase(tests.AdminTests):
+class ProjectAdminTestCase(tests.AdminTests, tests.AdminBaseTestCase):
     model_admin = admin.ProjectAdmin
     model = models.Project
     object_name = 'project'
@@ -50,7 +50,7 @@ class ProjectAdminTestCase(tests.AdminTests):
             name="new_name")
 
 
-class TaskAdminTestCase(tests.ReadOnlyAdminTests):
+class TaskAdminTestCase(tests.ReadOnlyAdminTests, tests.AdminBaseTestCase):
     """ Tests for read-only task admin."""
     model_admin = admin.TaskAdmin
     model = models.Task

